client/landing/stepper/utils/performance-tracking.tsx

```diff
@@ -0,0 +1,53 @@
+import { useLayoutEffect, useMemo } from 'react';
+import { useStore } from 'react-redux';
+import {
+	startPerformanceTracking,
+	stopPerformanceTracking,
+} from 'calypso/lib/performance-tracking/lib';
+
+// Manually start performance tracking.
+// Used in the entry point to start data collection as close as possible to boot time.
+export function startStepperPerformanceTracking( {
+	fullPageLoad = false,
+}: {
+	fullPageLoad?: boolean;
+} ) {
+	startPerformanceTracking( 'stepper', { fullPageLoad } );
+}
+
+// This hook starts performance gathering for the Stepper everytime the flow or step change.
+// The library we're using ignores follow-up `start` calls while there's an inflight tracking
+// report, so this hook is always safe, even though the entry point starts its own report on
+// first render.
+export function useStartStepperPerformanceTracking( flow: string, step: string ) {
+	useMemo( () => {
+		startStepperPerformanceTracking( { fullPageLoad: false } );
+		// We need to start tracking again everytime we change flow or step.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ flow, step ] );
+}
+
+function useStepperPerformanceTrackerStop( flow: string, step: string ) {
+	const performanceMetadata = useMemo( () => ( { flow: flow, step: step } ), [ flow, step ] );
+	const store = useStore();
+
+	// Use `useLayoutEffect` + rAF to be as close as possible to the actual rendering.
+	useLayoutEffect( () => {
+		requestAnimationFrame( () => {
+			stopPerformanceTracking( 'stepper', {
+				/// @ts-expect-error State is not properly typed in the performance tracking lib.
+				state: store.getState(),
```

Comment:
What do we want to capture from the store? I think we are not using the store.
The new flows rely more on the use of react-query than the store.

Reply:
This is the data that gets captured from the store:

```js
report.data.set( 'siteId', siteId );
report.data.set( 'siteIsJetpack', siteIsJetpack );
report.data.set( 'siteIsSingleUser', siteIsSingleUser );
report.data.set( 'siteIsAtomic', siteIsAtomic );
report.data.set( 'sitesCount', sitesCount );
report.data.set( 'sitesVisibleCount', sitesVisibleCount );
report.data.set( 'userCountryCode', userCountryCode );
report.data.set( 'userBootstrapped', userBootstrapped );
report.data.set( 'userLocale', userLocale );
report.data.set( 'translationsChunksDuration', translationsChunksDuration );
report.data.set( 'translationsChunksCount', translationsChunksCount );
```

I don't know which of those are relevant for Stepper purposes, but at least some of them do get filled in with non-trivial values. I'm happy to remove the store connection if this data isn't relevant.

Reply:
What do you think about starting with the minimum possible and then evolving? 

Minimum 
```javascript
report.data.set( 'siteId', siteId );  // We can filter a specific flow based on the siteId
report.data.set( 'userLocale', userLocale );  //  We can filter by locale
report.data.set( 'userCountryCode', userCountryCode ); // We can filter by country
```

The other ones seem interesting but the flows are very specialized and most of them are just not loading most of the required detail, like the number of sites or sites visible.


Do you agree?
cc:  @omarjackman @daledupreez

Reply:
What do you think about starting with the minimum possible and then evolving? 

Minimum 
```javascript
report.data.set( 'siteId', siteId );  // We can filter a specific flow based on the siteId
report.data.set( 'userLocale', userLocale );  //  We can filter by locale
report.data.set( 'userCountryCode', userCountryCode ); // We can filter by country
```

The other ones seem interesting but the flows are very specialized and most of them are just not loading most of the required detail, like the number of sites or sites visible.


Do you agree?
cc:  @omarjackman @daledupreez

Reply:
We can definitely do that, but in that case we won't be able to use `calypso/lib/performance-tracking/lib`'s `stopPerformanceTracking` any more, and will need to write our own.

Reply:
@gabrielcaires : Did you mean to mention @alshakero instead?

Reply:
Yes, sorry an thanks ! :)